src/content/learn/extracting-state-logic-into-a-reducer.md

```diff
@@ -179,17 +179,17 @@ li {
 
 </Sandpack>
 
-Each of its event handlers calls `setTasks` in order to update the state. As this component grows, so does the amount of state logic sprinkled throughout it. To reduce this complexity and keep all your logic in one easy-to-access place, you can move that state logic into a single function outside your component, **called a "reducer".**
+Setiap *event handler* pada komponen `TaskApp` akan memangil fungsi `setTask` untuk melakukan pembaharuan *state*. Saat komponen ini bertumbuh semakin besar, logika *state* akan semakin rumit. Untuk mengurangi kompleksitas dan menyimpan semua logika Anda di satu tempat yang mudah diakses. Anda dapat memindahkan logika *state* tersebut ke sebuah fungsi di luar komponen Anda, yang **disebut *"reducer"*.**
 
-Reducers are a different way to handle state. You can migrate from `useState` to `useReducer` in three steps:
+_Reducer_ merupakan sebuah alternatif untuk menangani *state*. Anda dapat migrasi dari `useState` ke `useReducer` dalam tiga langkah:
 
-1. **Move** from setting state to dispatching actions.
-2. **Write** a reducer function.
-3. **Use** the reducer from your component.
+1. **Pindah** dari memanggil fungsi penetap *state* menjadi aksi *dispatch*.
+2. **Tulis** fungsi *reducer*.
+3. **Gunakan** *reducer* pada komponen Anda.
```

Comment:
```suggestion
1. **Pindah** dari menyetel *state* menjadi men-*dispatch* *action*
2. **Tulis** fungsi *reducer*.
3. **Gunakan** *reducer* pada komponen Anda.
```

"aksi dispatch" sounds like a noun, where in this case it's supposed to be a verb, so "men-*dispatch*" sounds better.

And...well, *actions* and *dispatch* are also React*-specific pattern right? I guess we can leave it untranslated?

(*and yes, some other libraries also follow the same concept, but in general it's a very specific technical term to web development)

@mazipan @mhaidarhanif @zainfathoni